### DISCUSSION TITLE
Tracing Kokkos functions

### DISCUSSION THREAD

**Discussion by cwschilly** [LLNL/Caliper#552] [Q&A] [answered]
### Background
I profiling an application that uses Kokkos. Rather than adding annotations manually, I just set the `KOKKOS_TOOLS_LIBS` to point to my Caliper installation.

I've created a `caliper.config` file to set up custom configurations for profiling.

### Question
I'm looking for a way to trace Kokkos functions. Essentially, I want the same output as `export CALI_CONFIG="event-trace(trace.mpi)"`, but with Kokkos functions in addition to the MPI functions.

Here's my current MPI tracing config file:

```
CALI_SERVICES_ENABLE=event,mpi,recorder,timestamp,trace
CALI_EVENT_TRIGGER=function,mpi.function
CALI_TIMER_SNAPSHOT_DURATION=false
CALI_TIMER_INCLUSIVE_DURATION=false
CALI_TIMER_OFFSET=true
CALI_MPI_MSG_TRACING=true
CALI_MPI_BLACKLIST=MPI_Comm_rank,MPI_Comm_size,MPI_Wtime
CALI_RECORDER_FILENAME=trace-%mpi.rank%.cali
```

How can I alter this so that the Kokkos functions are also traced and recorded in the same `trace-*.cali` file?

**Comment by daboehme** [comment, marked as answer]
Hi @cwschilly, add `kokkostime` to the `CALI_SERVICES_ENABLE` list and remove the `CALI_EVENT_TRIGGER` line. That should do it. 

I can also add an option for the event-trace config to enable kokkos tracing.

**Reply by cwschilly** [reply]
Works perfectly, thanks!
